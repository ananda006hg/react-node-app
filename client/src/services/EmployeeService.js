import axios from 'axios';

const API_URL = 'http://localhost:5001/api/employees';

class EmployeeService {
  // Get all employees
  getAllEmployees() {
    return axios.get(API_URL);
  }

  // Get a single employee
  getEmployee(id) {
    return axios.get(`${API_URL}/${id}`);
  }

  // Create a new employee
  createEmployee(employee) {
    return axios.post(API_URL, employee);
  }

  // Update an employee
  updateEmployee(id, employee) {
    return axios.patch(`${API_URL}/${id}`, employee);
  }

  // Delete an employee
  deleteEmployee(id) {
    return axios.delete(`${API_URL}/${id}`);
  }
}

export default new EmployeeService();