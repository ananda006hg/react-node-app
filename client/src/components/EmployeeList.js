import React, { useState, useEffect } from 'react';
import { Table, Button, Container, Row, Col, Form, Modal, Alert } from 'react-bootstrap';
import { FaEdit, FaTrash, FaPlus } from 'react-icons/fa';
import EmployeeService from '../services/EmployeeService';

const EmployeeList = () => {
  const [employees, setEmployees] = useState([]);
  const [showModal, setShowModal] = useState(false);
  const [currentEmployee, setCurrentEmployee] = useState({
    name: '',
    email: '',
    position: '',
    phone: '',
    department: ''
  });
  const [isEditing, setIsEditing] = useState(false);
  const [searchTerm, setSearchTerm] = useState('');
  const [alert, setAlert] = useState({ show: false, message: '', variant: 'success' });

  // Fetch employees on component mount
  useEffect(() => {
    fetchEmployees();
  }, []);

  // Fetch all employees
  const fetchEmployees = () => {
    console.log('Fetching employees...');
    EmployeeService.getAllEmployees()
      .then(response => {
        console.log('Employees fetched:', response.data);
        setEmployees(response.data);
      })
      .catch(error => {
        console.error('Error fetching employees', error);
        showAlert('Error fetching employee data', 'danger');
      });
  };

  // Handle modal close
  const handleClose = () => {
    setShowModal(false);
    setCurrentEmployee({
      name: '',
      email: '',
      position: '',
      phone: '',
      department: ''
    });
    setIsEditing(false);
  };

  // Handle modal open for adding new employee
  const handleShow = () => {
    setShowModal(true);
    setIsEditing(false);
  };

  // Handle edit employee
  const handleEdit = (employee) => {
    setCurrentEmployee(employee);
    setIsEditing(true);
    setShowModal(true);
  };

  // Handle input change
  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setCurrentEmployee({ ...currentEmployee, [name]: value });
  };

  // Handle employee form submission
  const handleSubmit = (e) => {
    e.preventDefault();
    
    if (!currentEmployee.name || !currentEmployee.email || !currentEmployee.position) {
      showAlert('Please fill in all required fields', 'danger');
      return;
    }

    if (isEditing) {
      EmployeeService.updateEmployee(currentEmployee._id, currentEmployee)
        .then((response) => {
          console.log('Update response:', response);
          fetchEmployees();
          handleClose();
          showAlert('Employee updated successfully', 'success');
        })
        .catch(error => {
          console.error('Error updating employee', error);
          showAlert(`Error updating employee: ${error.response?.data?.message || error.message}`, 'danger');
        });
    } else {
      console.log('Saving employee:', currentEmployee);
      EmployeeService.createEmployee(currentEmployee)
        .then((response) => {
          console.log('Create response:', response);
          fetchEmployees();
          handleClose();
          showAlert('Employee added successfully', 'success');
        })
        .catch(error => {
          console.error('Error adding employee:', error);
          showAlert(`Error adding employee: ${error.response?.data?.message || error.message}`, 'danger');
        });
    }
  };

  // Handle employee deletion
  const handleDelete = (id) => {
    if (window.confirm('Are you sure you want to delete this employee?')) {
      EmployeeService.deleteEmployee(id)
        .then(() => {
          fetchEmployees();
          showAlert('Employee deleted successfully', 'success');
        })
        .catch(error => {
          console.error('Error deleting employee', error);
          showAlert('Error deleting employee', 'danger');
        });
    }
  };

  // Handle search filtering
  const filteredEmployees = employees.filter(employee =>
    employee.name.toLowerCase().includes(searchTerm.toLowerCase()) ||
    employee.position.toLowerCase().includes(searchTerm.toLowerCase()) ||
    employee.department?.toLowerCase().includes(searchTerm.toLowerCase())
  );

  // Show alert message
  const showAlert = (message, variant) => {
    setAlert({
      show: true,
      message,
      variant
    });
    
    setTimeout(() => {
      setAlert({ show: false, message: '', variant: 'success' });
    }, 3000);
  };

  return (
    <Container className="mt-4">
      {/* Alert message */}
      {alert.show && (
        <Alert variant={alert.variant}>
          {alert.message}
        </Alert>
      )}

      <Row className="mb-4">
        <Col md={6}>
          <h2>Employee Management</h2>
        </Col>
        <Col md={3}>
          <Form.Control 
            type="text" 
            placeholder="Search employees..." 
            value={searchTerm}
            onChange={(e) => setSearchTerm(e.target.value)}
          />
        </Col>
        <Col md={3} className="text-right">
          <Button variant="primary" onClick={handleShow}>
            <FaPlus /> Add New Employee
          </Button>
        </Col>
      </Row>

      {/* Employee Table */}
      <Table striped bordered hover responsive>
        <thead className="bg-dark text-white">
          <tr>
            <th>Name</th>
            <th>Email</th>
            <th>Position</th>
            <th>Phone</th>
            <th>Department</th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          {filteredEmployees.length > 0 ? (
            filteredEmployees.map(employee => (
              <tr key={employee._id}>
                <td>{employee.name}</td>
                <td>{employee.email}</td>
                <td>{employee.position}</td>
                <td>{employee.phone}</td>
                <td>{employee.department}</td>
                <td>
                  <Button 
                    variant="info" 
                    size="sm" 
                    className="mr-2"
                    onClick={() => handleEdit(employee)}
                  >
                    <FaEdit />
                  </Button>
                  {' '}
                  <Button 
                    variant="danger" 
                    size="sm"
                    onClick={() => handleDelete(employee._id)}
                  >
                    <FaTrash />
                  </Button>
                </td>
              </tr>
            ))
          ) : (
            <tr>
              <td colSpan="6" className="text-center">No employees found</td>
            </tr>
          )}
        </tbody>
      </Table>

      {/* Add/Edit Employee Modal */}
      <Modal show={showModal} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>{isEditing ? 'Edit Employee' : 'Add New Employee'}</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Form onSubmit={handleSubmit}>
            <Form.Group className="mb-3">
              <Form.Label>Name *</Form.Label>
              <Form.Control 
                type="text" 
                name="name"
                value={currentEmployee.name} 
                onChange={handleInputChange}
                required 
              />
            </Form.Group>
            
            <Form.Group className="mb-3">
              <Form.Label>Email *</Form.Label>
              <Form.Control 
                type="email" 
                name="email"
                value={currentEmployee.email} 
                onChange={handleInputChange}
                required 
              />
            </Form.Group>
            
            <Form.Group className="mb-3">
              <Form.Label>Position *</Form.Label>
              <Form.Control 
                type="text" 
                name="position"
                value={currentEmployee.position} 
                onChange={handleInputChange}
                required 
              />
            </Form.Group>
            
            <Form.Group className="mb-3">
              <Form.Label>Phone</Form.Label>
              <Form.Control 
                type="text" 
                name="phone"
                value={currentEmployee.phone} 
                onChange={handleInputChange}
              />
            </Form.Group>
            
            <Form.Group className="mb-3">
              <Form.Label>Department</Form.Label>
              <Form.Control 
                type="text" 
                name="department"
                value={currentEmployee.department} 
                onChange={handleInputChange}
              />
            </Form.Group>
            
            <div className="d-flex justify-content-end">
              <Button variant="secondary" className="me-2" onClick={handleClose}>
                Cancel
              </Button>
              <Button variant="primary" type="submit">
                {isEditing ? 'Update' : 'Save'}
              </Button>
            </div>
          </Form>
        </Modal.Body>
      </Modal>
    </Container>
  );
};

export default EmployeeList;