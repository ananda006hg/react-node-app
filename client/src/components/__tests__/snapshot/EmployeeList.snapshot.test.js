import React from 'react';
import { render } from '@testing-library/react';
import EmployeeList from '../../EmployeeList';
import EmployeeService from '../../../services/EmployeeService';

// Mock the EmployeeService
jest.mock('../../../services/EmployeeService');

describe('EmployeeList Snapshot Tests', () => {
  // Sample employee data
  const mockEmployees = [
    {
      _id: '1',
      name: 'John Doe',
      email: 'john@example.com',
      position: 'Developer',
      phone: '123-456-7890',
      department: 'Engineering'
    },
    {
      _id: '2',
      name: 'Jane Smith',
      email: 'jane@example.com',
      position: 'Designer',
      phone: '987-654-3210',
      department: 'Design'
    }
  ];

  test('should match empty state snapshot', () => {
    // Mock API to return empty array
    EmployeeService.getAllEmployees.mockResolvedValue({ data: [] });
    
    const { asFragment } = render(<EmployeeList />);
    
    // Capture the initial render snapshot
    expect(asFragment()).toMatchSnapshot();
  });
  
  test('should match snapshot with employees', async () => {
    // Mock API to return employee data
    EmployeeService.getAllEmployees.mockResolvedValue({ data: mockEmployees });
    
    // Use RTL's async utilities to render with the resolved data
    const { asFragment, findByText } = render(<EmployeeList />);
    
    // Wait for an element that appears after data loads
    await findByText('John Doe');
    
    // Capture the loaded state snapshot
    expect(asFragment()).toMatchSnapshot();
  });
  
  test('should match modal snapshot when adding an employee', async () => {
    // Mock API
    EmployeeService.getAllEmployees.mockResolvedValue({ data: mockEmployees });
    
    // Render component
    const { asFragment, findByText, getByText } = render(<EmployeeList />);
    
    // Wait for component to load
    await findByText('John Doe');
    
    // Open the add employee modal
    getByText(/Add New Employee/i).click();
    
    // Capture snapshot with modal open
    expect(asFragment()).toMatchSnapshot();
  });
});
