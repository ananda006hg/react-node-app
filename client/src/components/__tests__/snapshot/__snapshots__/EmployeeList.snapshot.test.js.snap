// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeList Snapshot Tests should match empty state snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-4 container"
  >
    <div
      class="mb-4 row"
    >
      <div
        class="col-md-6"
      >
        <h2>
          Employee Management
        </h2>
      </div>
      <div
        class="col-md-3"
      >
        <input
          class="form-control"
          placeholder="Search employees..."
          type="text"
          value=""
        />
      </div>
      <div
        class="text-right col-md-3"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
           Add New Employee
        </button>
      </div>
    </div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead
          class="bg-dark text-white"
        >
          <tr>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Position
            </th>
            <th>
              Phone
            </th>
            <th>
              Department
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="text-center"
              colspan="6"
            >
              No employees found
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EmployeeList Snapshot Tests should match modal snapshot when adding an employee 1`] = `
<DocumentFragment>
  <div
    class="mt-4 container"
  >
    <div
      class="mb-4 row"
    >
      <div
        class="col-md-6"
      >
        <h2>
          Employee Management
        </h2>
      </div>
      <div
        class="col-md-3"
      >
        <input
          class="form-control"
          placeholder="Search employees..."
          type="text"
          value=""
        />
      </div>
      <div
        class="text-right col-md-3"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
           Add New Employee
        </button>
      </div>
    </div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead
          class="bg-dark text-white"
        >
          <tr>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Position
            </th>
            <th>
              Phone
            </th>
            <th>
              Department
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              John Doe
            </td>
            <td>
              john@example.com
            </td>
            <td>
              Developer
            </td>
            <td>
              123-456-7890
            </td>
            <td>
              Engineering
            </td>
            <td>
              <button
                class="mr-2 btn btn-info btn-sm"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  />
                </svg>
              </button>
               
              <button
                class="btn btn-danger btn-sm"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </button>
            </td>
          </tr>
          <tr>
            <td>
              Jane Smith
            </td>
            <td>
              jane@example.com
            </td>
            <td>
              Designer
            </td>
            <td>
              987-654-3210
            </td>
            <td>
              Design
            </td>
            <td>
              <button
                class="mr-2 btn btn-info btn-sm"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  />
                </svg>
              </button>
               
              <button
                class="btn btn-danger btn-sm"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`EmployeeList Snapshot Tests should match snapshot with employees 1`] = `
<DocumentFragment>
  <div
    class="mt-4 container"
  >
    <div
      class="mb-4 row"
    >
      <div
        class="col-md-6"
      >
        <h2>
          Employee Management
        </h2>
      </div>
      <div
        class="col-md-3"
      >
        <input
          class="form-control"
          placeholder="Search employees..."
          type="text"
          value=""
        />
      </div>
      <div
        class="text-right col-md-3"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
           Add New Employee
        </button>
      </div>
    </div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-striped table-bordered table-hover"
      >
        <thead
          class="bg-dark text-white"
        >
          <tr>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Position
            </th>
            <th>
              Phone
            </th>
            <th>
              Department
            </th>
            <th>
              Actions
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              John Doe
            </td>
            <td>
              john@example.com
            </td>
            <td>
              Developer
            </td>
            <td>
              123-456-7890
            </td>
            <td>
              Engineering
            </td>
            <td>
              <button
                class="mr-2 btn btn-info btn-sm"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  />
                </svg>
              </button>
               
              <button
                class="btn btn-danger btn-sm"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </button>
            </td>
          </tr>
          <tr>
            <td>
              Jane Smith
            </td>
            <td>
              jane@example.com
            </td>
            <td>
              Designer
            </td>
            <td>
              987-654-3210
            </td>
            <td>
              Design
            </td>
            <td>
              <button
                class="mr-2 btn btn-info btn-sm"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  />
                </svg>
              </button>
               
              <button
                class="btn btn-danger btn-sm"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
